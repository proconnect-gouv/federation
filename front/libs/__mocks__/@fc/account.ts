import React from 'react';

export const AccountProvider = jest.fn();

export const AccountContext = React.createContext(expect.any(Object));
