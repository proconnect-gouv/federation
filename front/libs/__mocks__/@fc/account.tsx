import React from 'react';

export const AccountProvider = jest.fn(({ children }) => (
  <div>
    <div>AccountProvider</div>
    <div>{children}</div>
  </div>
));

export const AccountContext = React.createContext(expect.any(Object));
