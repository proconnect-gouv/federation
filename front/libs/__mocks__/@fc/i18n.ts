export const I18nService = {
  initialize: jest.fn(),
};

export const t = jest.fn((key) => key);
