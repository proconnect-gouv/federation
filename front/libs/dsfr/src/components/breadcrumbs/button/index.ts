export * from './breadcrumbs-toggle.button';
