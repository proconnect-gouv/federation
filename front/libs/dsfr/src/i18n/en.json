{
  "DSFR.stepper.location": "Step {current} of {total}",
  "DSFR.stepper.nextStep": "Next step"
}
