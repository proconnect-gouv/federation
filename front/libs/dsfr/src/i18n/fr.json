{
  "DSFR.stepper.location": "Étape {current} sur {total}",
  "DSFR.stepper.nextStep": "Étape suivante",
  "DSFR.breadcrumbs.show": "Voir le fil d’Ariane",
  "DSFR.breadcrumbs.location": "vous êtes ici :"
}
