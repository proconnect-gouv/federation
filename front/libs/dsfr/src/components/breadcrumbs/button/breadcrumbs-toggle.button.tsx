import { useToggle } from 'usehooks-ts';

import { t } from '@fc/i18n';

interface BreadCrumbsToggleButtonProps {
  id: string;
}

export const BreadCrumbsToggleButton = ({ id }: BreadCrumbsToggleButtonProps) => {
  const [show, toggleShow] = useToggle(false);

  return (
    <button
      aria-controls={id}
      aria-expanded={show}
      className="fr-breadcrumb__button"
      onClick={toggleShow}>
      {t('DSFR.breadcrumbs.show')}
    </button>
  );
};

BreadCrumbsToggleButton.displayName = 'BreadCrumbsToggleButton';
