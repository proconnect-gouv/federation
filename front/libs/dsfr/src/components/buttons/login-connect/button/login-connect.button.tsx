import classnames from 'classnames';
import React from 'react';

import type { PropsWithOnClick } from '@fc/common';
import { t } from '@fc/i18n';

import type { ButtonTypes } from '../../../../enums';
import { ConnectTypes } from '../../../../enums';
import styles from './login-connect.button.module.scss';

interface LoginConnectButtonProps extends PropsWithOnClick {
  type: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({ connectType, onClick, type }: LoginConnectButtonProps) => {
    const loginLabel = t('DSFR.button.loginConnect.authenticateWith');
    return (
      <button
        className={classnames('fr-connect', {
          [styles['fr-connect--proconnect']]: connectType === ConnectTypes.PRO_CONNECT,
        })}
        type={type}
        onClick={onClick}>
        <span className="fr-connect__login">{loginLabel}</span>
        <span className="fr-connect__brand">{connectType}</span>
      </button>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
