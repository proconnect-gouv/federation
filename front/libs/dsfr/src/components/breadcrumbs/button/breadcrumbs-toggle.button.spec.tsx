import { fireEvent, render } from '@testing-library/react';
import { useToggle } from 'usehooks-ts';

import { t } from '@fc/i18n';

import { BreadCrumbsToggleButton } from './breadcrumbs-toggle.button';

describe('BreadCrumbsToggleButton', () => {
  // Given
  const uniqid = 'any-uniqid-mock';
  const toggleShowMock = jest.fn();

  beforeEach(() => {
    // Given
    jest.mocked(t).mockReturnValue('DSFR.breadcrumbs.show');
    jest
      .mocked(useToggle)
      .mockReturnValue([
        'any-toggle-value-mock' as unknown as boolean,
        toggleShowMock,
        expect.any(Function),
      ]);
  });

  it('should match snapshot', () => {
    // When
    const { container, getByText } = render(<BreadCrumbsToggleButton id={uniqid} />);
    const button = getByText('DSFR.breadcrumbs.show');

    // Then
    expect(button).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(button).toHaveAttribute('aria-controls', 'any-uniqid-mock');
    expect(button).toHaveAttribute('aria-expanded', 'any-toggle-value-mock');
    expect(button).toHaveAttribute('class', 'fr-breadcrumb__button');
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('DSFR.breadcrumbs.show');
  });

  it('should update the button visibility value on click', () => {
    // When
    const { getByText } = render(<BreadCrumbsToggleButton id={uniqid} />);
    const button = getByText('DSFR.breadcrumbs.show');
    fireEvent.click(button);

    // Then
    expect(toggleShowMock).toHaveBeenCalledOnce();
  });
});
