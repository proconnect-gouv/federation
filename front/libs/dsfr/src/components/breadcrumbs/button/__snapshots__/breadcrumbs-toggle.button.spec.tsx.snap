// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbsToggleButton should match snapshot 1`] = `
<div>
  <button
    aria-controls="any-uniqid-mock"
    aria-expanded="any-toggle-value-mock"
    class="fr-breadcrumb__button"
  >
    DSFR.breadcrumbs.show
  </button>
</div>
`;
